src/main/java/it/gov/pagopa/apiconfig/core/service/IbanService.java

```diff
@@ -0,0 +1,159 @@
+import it.gov.pagopa.apiconfig.core.exception.AppException;
+import it.gov.pagopa.apiconfig.core.model.creditorinstitution.IbanLabel;
+import it.gov.pagopa.apiconfig.core.model.creditorinstitution.IbanV2;
+import it.gov.pagopa.apiconfig.core.model.creditorinstitution.IbansV2;
+import it.gov.pagopa.apiconfig.core.util.CommonUtil;
+import it.gov.pagopa.apiconfig.starter.entity.Iban;
+import it.gov.pagopa.apiconfig.starter.entity.IbanAttribute;
+import it.gov.pagopa.apiconfig.starter.entity.IbanAttributeMaster;
+import it.gov.pagopa.apiconfig.starter.entity.IbanMaster;
+import it.gov.pagopa.apiconfig.starter.entity.IbanMaster.IbanStatus;
+import it.gov.pagopa.apiconfig.starter.entity.IcaBinaryFile;
+import it.gov.pagopa.apiconfig.starter.entity.Pa;
+import it.gov.pagopa.apiconfig.starter.repository.IbanAttributeMasterRepository;
+import it.gov.pagopa.apiconfig.starter.repository.IbanAttributeRepository;
+import it.gov.pagopa.apiconfig.starter.repository.IbanMasterRepository;
+import it.gov.pagopa.apiconfig.starter.repository.IbanRepository;
+import it.gov.pagopa.apiconfig.starter.repository.IcaBinaryFileRepository;
+import it.gov.pagopa.apiconfig.starter.repository.PaRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Validated
+@Transactional
+public class IbanService {
+
+  @Autowired private PaRepository paRepository;
+
+  @Autowired private IbanRepository ibanRepository;
+
+  @Autowired private IbanMasterRepository ibanMasterRepository;
+
+  @Autowired private IbanAttributeRepository ibanAttributeRepository;
+
+  @Autowired private IbanAttributeMasterRepository ibanAttributeMasterRepository;
+
+  @Autowired private IcaBinaryFileRepository icaBinaryFileRepository;
+
+  @Autowired private ModelMapper modelMapper;
+
+  public IbanV2 createIban(@NotBlank String organizationFiscalCode, @Valid @NotNull IbanV2 iban) {
+    // retrieve the creditor institution and throw exception if not found
+    Optional<Pa> creditorInstitutionOpt = paRepository.findByIdDominio(organizationFiscalCode);
+    Pa existingCreditorInstitution = creditorInstitutionOpt.orElseThrow(() -> new AppException(AppError.CREDITOR_INSTITUTION_NOT_FOUND, organizationFiscalCode));
+    // retrieve an existing iban or generate a new one if not defined
+    Iban ibanToBeCreated = ibanRepository.findByIban(iban.getIbanValue()).orElse(null);
+    if (ibanToBeCreated == null) {
```

Comment:
if the iban is a postal iban we need to check if the `oraginaztionFiscalCode` is that of the post office